lib/crypto/secp256k1/secp256k1.go

```diff
@@ -36,6 +37,17 @@ type PublicKey struct {
 	key ecdsa.PublicKey
 }
 
+// VerifySignature verifies a signature given a public key and a message
+func VerifySignature(publicKey, signature, message []byte) error {
+	ok := secp256k1.VerifySignature(publicKey, message, signature)
+	if !ok {
+		return fmt.Errorf("secp256k1: %w: for message 0x%x, signature 0x%x and public key 0x%x",
+			crypto.ErrSignatureVerificationFailed, message, signature, publicKey)
+	}
+
+	return nil
```

Comment:
nit you could flip those around to have less indentation / line length

```suggestion
	if ok {
		return nil
	}

	return fmt.Errorf("secp256k1: %w: for message 0x%x, signature 0x%x and public key 0x%x",
		crypto.ErrSignatureVerificationFailed, message, signature, publicKey)
```